weblate/templates/js/shortcut.js

```diff
@@ -0,0 +1,228 @@
+      if (e.shiftKey) modifiers.shift.pressed = true;
+      if (e.altKey) modifiers.alt.pressed = true;
+      if (e.metaKey) modifiers.meta.pressed = true;
+
+      for (var i = 0; (k = keys[i]), i < keys.length; i++) {
+        //Modifiers
+        if (k == "ctrl" || k == "control") {
+          kp++;
+          modifiers.ctrl.wanted = true;
+        } else if (k == "shift") {
+          kp++;
+          modifiers.shift.wanted = true;
+        } else if (k == "alt") {
+          kp++;
+          modifiers.alt.wanted = true;
+        } else if (k == "meta") {
+          kp++;
+          modifiers.meta.wanted = true;
+        } else if (k.length > 1) {
+          //If it is a special key
+          if (special_keys[k] == code) kp++;
+        } else if (opt["keycode"]) {
+          if (opt["keycode"] == code) kp++;
+        } else {
+          //The special keys did not match
+          if (character == k) kp++;
+          else {
+            if (shift_nums[character] && e.shiftKey) {
+              //Stupid Shift key bug created by using lowercase
+              character = shift_nums[character];
+              if (character == k) kp++;
+            }
+          }
+        }
+      }
+
+      if (
+        kp == keys.length &&
+        modifiers.ctrl.pressed == modifiers.ctrl.wanted &&
+        modifiers.shift.pressed == modifiers.shift.wanted &&
+        modifiers.alt.pressed == modifiers.alt.wanted &&
+        modifiers.meta.pressed == modifiers.meta.wanted
+      ) {
+        callback(e);
+
+        if (!opt["propagate"]) {
+          //Stop the event
+          //e.cancelBubble is supported by IE - this will kill the bubbling process.
+          e.cancelBubble = true;
+          e.returnValue = false;
+
+          //e.stopPropagation works in Firefox.
+          if (e.stopPropagation) {
+            e.stopPropagation();
+            e.preventDefault();
+          }
+          return false;
+        }
+      }
+    };
+    this.all_shortcuts[shortcut_combination] = {
+      callback: func,
+      target: ele,
+      event: opt["type"],
+    };
+    //Attach the function with the event
+    if (ele.addEventListener) ele.addEventListener(opt["type"], func, false);
+    else if (ele.attachEvent) ele.attachEvent("on" + opt["type"], func);
+    else ele["on" + opt["type"]] = func;
+  },
+
+  //Remove the shortcut - just specify the shortcut and I will remove the binding
+  remove: function (shortcut_combination) {
+    shortcut_combination = shortcut_combination.toLowerCase();
+    var binding = this.all_shortcuts[shortcut_combination];
+    delete this.all_shortcuts[shortcut_combination];
+    if (!binding) return;
+    var type = binding["event"];
+    var ele = binding["target"];
+    var callback = binding["callback"];
+
+    if (ele.detachEvent) ele.detachEvent("on" + type, callback);
+    else if (ele.removeEventListener)
+      ele.removeEventListener(type, callback, false);
+    else ele["on" + type] = false;
+  },
+};
```

Comment:
This should not be needed, we already have bundled Mousetrap library for keyboard shortcuts.